LiteCore/Query/QueryParser.cc

```diff
@@ -1132,6 +1170,91 @@ namespace litecore {
         }
     }
 
+    namespace {
+        slice const kMetaKeys[] = {
+            "id"_sl,
+            "sequence"_sl,
+            "deleted"_sl,
+            "expiration"_sl,
+            "revisionID"_sl
+        };
+        enum {
+            mkId,
+            mkSequence,
+            mkDeleted,
+            mkExpiration,
+            mkRevisionId,
+            mkCount
+        };
+        static_assert(sizeof(kMetaKeys) / sizeof(kMetaKeys[0]) == mkCount);
+    }
+
+    // Handles ["meta", dbAlias_optional]
+    void QueryParser::metaOp(slice op, Array::iterator& operands) {
+        // Pre-conditions: op == "meta"
+        //                 operands.size() == 0 || operands[0]->type() == kString (dbAlias)
+
+        string arg;
+        if (operands.count() > 0 && operands[0]->type() == kString) {
+            arg = operands[0]->asString();
+        }
+        
+        Path path {slice(arg+".id")};
+        const auto& dbIter = verifyDbAlias(path);
+        require(arg.empty() || arg == dbIter->first,
+                "database alias '%s' does not match a declared 'AS' alias", arg.c_str());
```

Comment:
Looks like these lines could go inside the `if{}` block above, since they only apply if there is an operand.

Reply:
Good advice and I will do it.

Reply:
Oh, wait, cannot do that. The above if{} is to get the optional argument. Regardless of the presence of the optional argument, we have to run the following piece of code.